docs/adr/2022-04-26__alerting.md

```diff
@@ -0,0 +1,295 @@
+# Alerting in OpMon
+
+* Author: Anna Scholtz
+* Date: 2022-04-26
+* Status: Accepting Feedback (Feedback from DS is especially welcome)
+
+## Introduction
+
+Users of OpMon want to be able to see critical changes and optionally get notified whenever they occur. This ADR describes how OpMon will provide the ability to define different monitors, how they will be implemented and how users can subscribe to getting email notifications.
+
+As an example, of what the operational monitoring dashboards currently look like see: https://mozilla.cloud.looker.com/dashboards/operational_monitoring::bug_1715474_rollout_yandex_sponsored_tile_rollout_release_89_100?Percentile=50
+The dashboards have a tile for each metric which shows the values of all available branches for the percentile currently selected as filter.
+
+## Constraints
+
+There are a couple of constraints when checking for critical changes in OpMon:
+* **Cost:** We will want to keep the cost low since there will be a lot of dashboards (for example, we create one for each rollout)
+* **Computation Speed:** The operational monitoring dashboards get updated on a daily basis and there are usually dozens of dashboards that need to get updated. Ideally, the checks for each dashboard and day finish within minutes rather than hours.
+* **Minimize False Positives:** Sending alerts when nothing is wrong too many times results in users simply ignoring alerts at some point
+
+## Monitor Types
+
+When talking to users there are a few different cases when they would like to receive an alert:
+* **Large differences between branches:** Whenever the confidence intervals of different branches for a specific metric no longer overlap, it indicates that there is potentially some significant difference.
+* **Thresholds:** Comparing the values of a metric to a user-defined threshold.
+* **Deviation from historical data:** Detect anomalous behaviour of a metric based on previously collected data.
+
+### Large differences between branches
+
+The OpMon dashboards show the values for specific metrics as a line chart with confidence intervals. Each line represents the metric values for a different branch. Whenever the confidence intervals of the branches do not overlap, it is considered a critical change. See:
```

Comment:
`Whenever the confidence intervals of the branches do not overlap, it is considered a critical change.` Statistically significant changes can also occur when the confidence intervals overlap slightly. If two 95% confidence intervals don't overlap, we have more than 95% confidence that there is a difference (with the degree of confidence depending on the correlation between the two metrics). 

Since critical changes can occur when CIs overlap slightly, perhaps we should come up with another way of visually communicating that there's a change such as highlighting when the difference is statistically significant (again, a less strict criteria than when the CIs overlap).

Reply:
That makes sense. I guess one thing I'm concerned if we loosen the criteria here, as in there is a change if confidence intervals either do not overlap or they can overlap slightly, it becomes more difficult to define what _slightly_ means in that case.
Not sure if you have a good method/indicators what it means for the difference to be statistically significant?

Reply:
In this situation we usually measure statistical significance using a [2-sample t-test ](https://en.wikipedia.org/wiki/Student%27s_t-test#Independent_two-sample_t-test). I'd say it's not that we're loosening the criteria, it's more that by using "do the 95% confidence intervals overlap" as our criteria, that's more strict than 95%. So there may be situations where we miss a significant difference that would have otherwise triggered at 95%. 

It's not that this approach is bad, it's just a more strict criteria than we might believe so we should communicate that to the users.